lib/index.js

```diff
@@ -61,7 +61,7 @@ module.exports = (options = {}) => {
     const encodings = new Encodings({
       preferredEncodings
     })
-    encodings.parseAcceptEncoding(ctx.request.headers['accept-encoding'] || undefined)
+    encodings.parseAcceptEncoding(ctx.request.headers['accept-encoding'] || 'identity')
```

Comment:
the default is not actually `identity` but `*`. https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Encoding#Directives

> `*` Matches any content encoding not already listed in the header. This is the default value if the header is not present. It doesn't mean that any algorithm is supported; merely that no preference is expressed.

Reply:
i'm happy to make a default option here though

Reply:
I found it *really* confusing when I used curl to hit my server and got back compressed responses even though I didn't ask for them.

Even if the spec says that the default is "don't care" I think on the server side it's better to play it safe and only use compression formats explicitly supported by the client.

I think HTTP clients that support compression will almost always send the header, and ones that do not support compression will probably almost never send it, so it makes more sense to me that way (assume `identity`).

Anyway, I've made my case :)  Do with it what you will.

Reply:
i agree, that spec is confusing

Reply:
@jonathanong 

FWIW, this does seem to be a different behavior than was present in v3 of this library.  Since the previous version of the library used koa's `ctx.acceptsEncodings`, it means this behavior differs from koa (and for that matter from express).  So I would assume the confusion largely comes from that.

The previous version ultimately used `negotiator`, which always adds `identity` to the list of accepted encodings: https://github.com/jshttp/negotiator/blob/master/lib/encoding.js#L46-L56

If the new behavior is more correct according to the spec, maybe its worth explicitly calling that out since its a departure from previous versions?